### DISCUSSION TITLE
EBNF: one_or_more_with_separators

### DISCUSSION THREAD

**Discussion by AndreasMatthias** [plantuml/plantuml#1452] [Ideas]
Plantuml displays "one or more with separator" in EBNF diagrams like this:

```
  @startebnf
  bar = foo, {",", foo};
  !pragma compact
  bar = foo, {",", foo};
  @endebnf
```
![plantuml](https://github.com/plantuml/plantuml/assets/1072202/eb8d80d3-6985-4e5d-9a13-b0d932e0e12d)

What I'm missing is the following representation:

![rr](https://github.com/plantuml/plantuml/assets/1072202/887ded26-2203-4985-a909-6b3dc01701a3)

**Comment by The-Lum** [comment]
Hi @AndreasMatthias, and all,

IMHO, this is a restriction of the current implementation:
- because with more than one element the backward notation is not so understandable.

_See order to `a` and `b` on the follow example:_

```

	    ╭━━━━━━━>━━━━━━━━━━━━━━━━━━━━━━━╮ 
            ┃  ╭━━━━━━━<━━━━━━━━━━━━━━━━━╮  ┃
  ┏━━━━━┓   ┃  ┃  ┏━━━┓ ┏━━━┓ ┏━━━━━┓    ┃  ┃
┣━┫ foo ┣━━━╯━━╰━━┫ a ┣━┫ b ┣━┫ foo ┣━━━━╯━━╰━━━> 
  ┗━━━━━┛         ┗━━━┛ ┗━━━┛ ┗━━━━━┛
  

       ┏━━━┓ ┏━━━┓
     ╭━┫ b ┣━┫ a ┣━<━╮
     ┃ ┗━━━┛ ┗━━━┛   ┃
     ┃   ┏━━━━━┓     ┃
┣━━━━╰━━━┫ foo ┣━━━━━╯━━━━━> 
         ┗━━━━━┛ 
```

_To debate..._
Regards.

**Reply by AndreasMatthias** [reply]
You are right, this representation can be quite confusing if there is **more than one** symbol in the loop-back. But if there's **only one** symbol in the loop-back, I think it's much easier to read. All lists with a single separator (like comma separated lists) would benefit from this representation.

If someone implements this representation I guess it would really be necessary to have another pragma for enabling/disabling this feature.

**Reply by toxn** [reply]
This could be implemented without `pragma`, with an alteration of the one-or-more-ebnf like this :
```
@startebnf
bar = {foo}-",";
@endebnf
```
